packages/ui/components/form-core/src/FocusMixin.js

```diff
@@ -52,13 +53,43 @@ const FocusMixinImplementation = superclass =>
     connectedCallback() {
       super.connectedCallback();
       this.__registerEventsForFocusMixin();
+      this.__syncAutofocusToFocusableElement();
     }
 
     disconnectedCallback() {
       super.disconnectedCallback();
       this.__teardownEventsForFocusMixin();
     }
 
+    /**
+     * Gets called when an attribute is changed.
+     * @param {String} name
+     * @param {String | null} _old
+     * @param {String | null} value
+     * @protected
+     */
+    attributeChangedCallback(name, _old, value) {
+      super.attributeChangedCallback(name, _old, value);
+      if (name === 'autofocus') {
+        this.__syncAutofocusToFocusableElement();
+      }
+    }
+
+    /**
+     * @private
+     */
+    __syncAutofocusToFocusableElement() {
+      if (!this._focusableNode) {
+        return;
+      }
+
+      if (this.hasAttribute('autofocus')) {
+        this._focusableNode.setAttribute('autofocus', '');
```

Comment:
Hey, nice PR :)

Could this be tested in an integration test with our dialog system as well?

The spec says:
>No more than one element in the document or dialog may have the autofocus attribute. If  applied to multiple elements the first one will receive focus.

https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autofocus

Since recently we [use a native `<dialog>`](https://github.com/ing-bank/lion/pull/1712) under the hood.
I suspect that if both the host and _focusableNode have focus, the platform might attempt to put focus on the parent element (in case we don't configure the OverlayController with `trapsKeyboardFocus: true`, in which case [it calls contain-focus, like mentioned here](https://github.com/ing-bank/lion/pull/1776#event-9007140697))

Could we rule this out via an integration test in the overlay system?

Reply:
Hi Thijs, good point. I've tried to write a test for what happens if "trapsKeyboardFocus: false" is set. 
The good news is that it won't focus the host. The potentially bad news is that it will not focus the _focusableNode either. 

I've done some manual testing on the desktop versions of Firefox, Chromium and Safari and found the following. In all cases I tested with autofocus synced: 

1. Test case: A LionDialog with trapsKeyboardFocus set to true and an LionCheckbox with autofocus. Result: The focusable node was focused.
2. Test

If trapsKeyboardFocus is set to true autofocus works as intended. If trapsKeyboardFocus is set to false autofocus will not work, pressing tab once will set the focus to the first focusable element in the dialog. 
This holds true both for a LionField with autofocus or a native html input with autofocus. Using a LionField with autofocus in a native dialog will focus the focusable_node, not the host

Reply:
Yes indeed for nr 2. Its not a new page, so the focus stays where it is.

I've added some focus tests to the dialog and form-integration.

Reply:
Hey @Sciurus7 , thanks for all the great help and testing this. Sorry that I postponed my repsonse (wanted to test some things in the browser but didn't get to it...)
Luckily Gerjan took over so I'm happy to merge this :)